app/router/router.go

```diff
@@ -0,0 +1,116 @@
+package router
+
+import (
+	"regexp"
+	"sort"
+	"strconv"
+
+	"github.com/lbryio/lbrytv/config"
+	"github.com/lbryio/lbrytv/internal/monitor"
+)
+
+type SDKRouter struct {
+	LbrynetServers map[string]string
+	logger         monitor.ModuleLogger
+	servers        []SDK
+}
+
+//SDK represents the Name and Address of an SDK server location
+type SDK struct { //Remove this once we can store names in DB
+	Name    string
+	Address string
+}
+
+func SingleSDKSet(address string) map[string]string {
+	return map[string]string{"default": address}
+}
+
+func New(sdkServers map[string]string) SDKRouter {
+	return SDKRouter{
+		LbrynetServers: sdkServers,
+		logger:         monitor.NewModuleLogger("router"),
+	}
+}
+
+func NewDefault() SDKRouter {
+	return SDKRouter{LbrynetServers: map[string]string{
+		"default": config.GetLbrynet(),
+	},
+		logger: monitor.NewModuleLogger("router"),
+	}
+}
+
+func (r *SDKRouter) GetSDKServerList() []SDK {
+	if len(r.servers) == 0 {
+		r.sortServers()
+	}
+	return r.servers
+}
+
+func (r *SDKRouter) GetSDKServer(walletID string) string {
+	sdk := r.balancedSDK()
+	if walletID != "" {
+		sdk = r.getSDKByWalletID(walletID)
+	}
+
+	if walletID != "" && sdk == "" {
+		r.logger.Log().Info("walletID is set but there is no server associated with it.")
+		sdk = r.defaultSDKServer() //"UNKOWN_SERVER"
+	}
+	r.logger.Log().Infof("From wallet id [%s] server [%s] is returned", walletID, sdk)
+	return sdk
+}
+
+func (r *SDKRouter) getSDKByWalletID(walletID string) string {
+	if len(r.servers) == 0 {
```

Comment:
Why don't we sort them servers once in the `New()` and only check here the list length for sanity?